conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/NegationRuleInterpreter.scala

```diff
@@ -40,8 +40,10 @@ case class NegationRuleInterpreter(rule: NegationConformanceRule) extends RuleIn
       case _: DecimalType =>
         // Negating decimal cannot fail
         DeepArrayTransformations.nestedWithColumnMap(df, rule.inputColumn, rule.outputColumn, c => negate(c))
-      case _: DoubleType | _: FloatType =>
-        // Negating floating point numbers cannot fail, but we need to account for signed zeros (see the note for getNegator()).
+      case _: DoubleType | _: FloatType | _: BooleanType =>
+        // Negating boolean values and floating point numbers cannot fail. But:
```

Comment:
The evaluation is already somewhat chaotic, and repeatedly test for `fieldType`. Not requesting a full refactor, but could you perhaps make the `BooleanType` a separate `case` similarly to `DecimalType` with the `not(c)` used directly?

Reply:
Refactored a little. Please check